lib/cache/add-remote-git.js

```diff
@@ -74,40 +85,20 @@ function addRemoteGit (uri, _cb) {
   }
 }
 
-function tryGitProto (from, hostedInfo, cb) {
-  var gitURL = hostedInfo.git()
-  if (!gitURL) return tryHTTPS(from, hostedInfo, cb)
-
-  log.silly('tryGitProto', 'attempting to clone', gitURL)
-  tryClone(from, gitURL, true, function (er) {
-    if (er) return tryHTTPS(from, hostedInfo, cb)
-
-    cb.apply(this, arguments)
-  })
-}
-
-function tryHTTPS (from, hostedInfo, cb) {
-  var httpsURL = hostedInfo.https()
-  if (!httpsURL) {
-    return cb(new Error(from + ' can not be cloned via Git, SSH, or HTTPS'))
+function tryProtos (from, services, previousError, cb) {
+  if (services.length === 0) {
+    return cb(previousError ||
+              new Error(from + ' can not be cloned via HTTPS, SSH or Git'))
   }
-
-  log.silly('tryHTTPS', 'attempting to clone', httpsURL)
-  tryClone(from, httpsURL, true, function (er) {
-    if (er) return trySSH(from, hostedInfo, cb)
-
+  var service = services.shift()
+  var silent = (service.protocol !== 'ssh')
+  log.silly('tryProtos', 'attempting to clone', url)
```

Comment:
Sure, thanks! I used to have just `url` and `urls` instead of `service` and `services`, but then switched to the latter to write the `silent` flag without having to parse the URL again, and forgot this line. Will amend.